src/backup/on_demand/on_demand_backup_predicate.py

```diff
@@ -2,17 +2,22 @@
 
 from src.backup.abstract_backup_predicate import \
     AbstractBackupPredicate
+from src.commons.exceptions import ParameterValidationException, NotFoundException
 
 
 class OnDemandBackupPredicate(AbstractBackupPredicate):
 
     def test(self, big_query_table_metadata, table_entity):
-        if not self._is_possible_to_copy_table(big_query_table_metadata):
-            return False
+        table_validation_status, table_validation_message = self._is_possible_to_copy_table(big_query_table_metadata)
+        if not table_validation_status:
+            if table_validation_message == "Table not found":
+                raise NotFoundException(table_validation_message)
+            else:
+                raise ParameterValidationException(table_validation_message)
 
```

Comment:
I would like to move validation about `"Partition id is required for partitioned table in on-demand mode`, which is currently done in OnDemandTableBackup class.